src/settings/mod.rs

```diff
@@ -1,28 +1,72 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, you can obtain one at https://mozilla.org/MPL/2.0/.
-
 //! Application settings.
 
-use std::env;
-
 use config::{Config, ConfigError, Environment, File};
+use serde::de::{Deserialize, Deserializer, Error, Unexpected};
+use std::{
+    env,
+    fmt::{self, Display},
+};
 
-use deserialize;
+use duration::Duration;
 use logging::MozlogLogger;
 use serialize;
+use validate;
 
 #[cfg(test)]
 mod test;
 
+macro_rules! deserialize {
```

Comment:
No biggie, but seeing as two of the arguments relate to validation I wonder if the macro should mention validation too? No worries if you think it's too long, but how do you feel about `deserialize_and_validate`?

Reply:
Perfect name, agreed.